server/resp/src/main/java/org/infinispan/server/resp/json/JsonClearFunction.java

```diff
@@ -0,0 +1,114 @@
+package org.infinispan.server.resp.json;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.JsonPath;
+import org.infinispan.commons.CacheException;
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.functional.EntryView.ReadWriteEntryView;
+import org.infinispan.server.resp.ExternalizerIds;
+import org.infinispan.util.function.SerializableFunction;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+public class JsonClearFunction
+      implements SerializableFunction<ReadWriteEntryView<byte[], JsonBucket>, Integer> {
+   public static final String ERR_PATH_CAN_T_BE_NULL = "path can't be null";
+   public static final AdvancedExternalizer<JsonClearFunction> EXTERNALIZER = new JsonClearFunction.Externalizer();
+
+   byte[] path;
+
+   public JsonClearFunction(byte[] path) {
+      requireNonNull(path, ERR_PATH_CAN_T_BE_NULL);
+      this.path = path;
+   }
+
+   @Override
+   public Integer apply(ReadWriteEntryView<byte[], JsonBucket> entryView) {
+      var doc = entryView.find().orElse(null);
+      if (doc == null) {
+         throw new CacheException("could not perform this operation on a key that doesn't exist");
+      }
+
+      var pathStr = new String(path, StandardCharsets.UTF_8);
+      try {
+         JsonNode rootNode = JSONUtil.objectMapper.readTree(doc.value());
+         DocumentContext modifiableCtx = JSONUtil.parserForMod.parse(rootNode);
+         DocumentContext getForContext = JSONUtil.parserForGet.parse(rootNode);
+         JsonPath jpath = JsonPath.compile(pathStr);
+         ArrayNode pathList = modifiableCtx.read(jpath);
+         int changed = 0;
+         for (JsonNode pathAsNode : pathList) {
+            String pathAsText = pathAsNode.asText();
+            ArrayNode node = getForContext.read(pathAsText);
+            if (JSONUtil.isRoot(pathAsText.getBytes(StandardCharsets.UTF_8))) {
+               // if the node is root, we need to replace all the value.
+               entryView.set(new JsonBucket(JSONUtil.objectMapper.writeValueAsBytes(JSONUtil.objectMapper.createObjectNode())));
```

Comment:
this replaces numeric root object with an empty object instead of 0

Reply:
I fixed it